datalad/support/gitrepo.py

```diff
@@ -3023,7 +3033,7 @@ def update_submodule(self, path, mode='checkout', init=False):
                 )
         # TODO: return value
 
-    def update_ref(self, ref, value, symbolic=False):
+    def update_ref(self, ref, value, oldvalue=None, symbolic=False):
```

Comment:
Repositioning `symbolic` is a breaking change but probably okay.  I don't see any spots in our code (core or extensions) that pass it as a positional argument, and it's probably not used in third-party code.  Then again, perhaps the more natural positioning of `oldvalue` after `value` isn't worth any potential breakage, and `oldvalue` should just be moved to the end.

(In my view it'd be good to use keyword-only arguments more for newly added functions to avoid these sorts of issues.  It looks like `_get_dot_git` is the only function in the code base that currently does.)

Reply:
Yeah I couldn't find a spot using `symbolic` positional either. My rational here was the other way around: Future change should most likely remove that - in my view weird - janus headed functionality and move symbolic-ref into its own function. If I had appended it, we would then encounter the situation where we'd remove an argument from inbetween.

Reply:
> and move symbolic-ref into its own function.

True, or just delete the method, as its only extra logic/handling on top of a straight `call_git` is looking at `symbolic` and deciding the command.

Anyway, your choice/rationale is fine by me.